### PR TITLE
add new settings to ui

### PR DESCRIPTION
**Pull request by jlmitch5**
This is for adding the follow settings to the UI:

- [x] AWX_ISOLATED_CHECK_INTERVAL
- [x] AWX_ISOLATED_LAUNCH_TIMEOUT
- [x] AWX_ISOLATED_CONNECTION_TIMEOUT

---

- [x] ACCESS_TOKEN_EXPIRES_SECONDS
- [x] AUTHORIZATION_CODE_EXPIRE_SECONDS

These two didn't really fit anywhere existing, so I created a new misc. tab under auth to house them.  Also, the help text is strange, because the API passes back a singular help text for both fields, so both have this as the body of the help text popover:

screen shot 2019-01-15 at 2 19 36 pm

I confirmed these two things were acceptable with @trahman73.

I also tested that saving single (and both) values, reseting a single value, and resetting all values worked with these two fields.

### CODE REVIEW

**Comment by wenottingham** [comment]
Is there a way to make those two settings not wrap like that?

**Comment by jlmitch5** [comment]
@wenottingham that's how all of these forms work:

screen shot 2019-01-15 at 2 00 27 pm

File: awx/ui/client/src/configuration/forms/auth-form/configuration-auth.controller.js [hunk 1]
```diff
@@ -198,11 +199,17 @@ export default [
...
+                formDef: formDefs.authMisc,
+                id: 'auth-misc-form',
+                name: 'authMisc'
+            },
         ];
         var forms = _.map(authForms, 'formDef');
         _.each(forms, function(form) {
             var keys = _.keys(form.fields);
             _.each(keys, function(key) {
+                console.log(key);
```
Comment (jakemcdermott) [inline awx/ui/client/src/configuration/forms/auth-form/configuration-auth.controller.js]: 👀

**Comment by rooftopcellist** [comment]
It's not great that we can't split the help_text.  But with what we get from django-oauth-toolkit, this is probably the best we will get unless we choose to manually pass help_text in the UI with javascript.

**Review by jakemcdermott — COMMENTED** [review]
The code changes look good to me - nice work.

I also gave it a run locally and did some rudimentary usage of the forms. Things seemed to work as expected.

@elyezer is on point for testing this so I will defer to him for the final PR approval.

**Comment by kialam** [comment]
@jlmitch5 
Since we already have a misc settings tab that contains various authentication preferences in Settings>System>Misc. System, I think it would make more sense to move those two settings, `ACCESS_TOKEN_EXPIRES_SECONDS` and `AUTHORIZATION_CODE_EXPIRE_SECONDS` there rather than to their own tab. Thoughts on this?

screen shot 2019-01-17 at 11 58 02 am

**Comment by elyezer** [comment]
I was playing with this and did the following:

* Check if the new options are present on the UI
* Check if updating the setting persists, even across different sessions
* Check if the option is really used where they should be used
* Check if using the API to change the values reflect on UI
* Do boundary testing to ensure max character length or number is validate properly

And all the above worked fine. I have some comments to improve the help messages of `ACCESS_TOKEN_EXPIRES_SECONDS` and `AUTHORIZATION_CODE_EXPIRE_SECONDS`. Both help messages state the following:

> Dictionary for customizing OAuth 2 timeouts, available items are `ACCESS_TOKEN_EXPIRE_SECONDS`, the duration of access tokens in the number of seconds, and `AUTHORIZATION_CODE_EXPIRE_SECONDS`, the duration of authorization grants in the number of seconds.

But there is no way to define dictionary on the UI and also it one mentions information about the other. I think the help messages could be improved and show only the information about the related field.

Also those fields are related to OAUTH2 and this is not completely clear, maybe the name of the option on the UI could mention that, or the updated help message.

**Comment by elyezer** [comment]
While playing with this PR I found that these settings suffers from https://github.com/ansible/awx/issues/3022 and https://github.com/ansible/awx/issues/3023

**Review by kialam — APPROVED** [review]
Great job!

**Pull request merged**